OpenRA.Mods.RA/Traits/Disguise.cs

```diff
@@ -49,10 +49,10 @@ public Player Owner
 		{
 			get
 			{
-				if (disguise.Disguised)
-					return self.Owner == self.World.LocalPlayer ? self.Owner : disguise.AsPlayer;
+				if (!disguise.Disguised || self.Owner.IsAlliedWith(self.World.RenderPlayer))
```

Comment:
Not really, because the radar color may be as wrong as it likes, it doesn't affect the game state at all.